autoemulate/experimental/emulators/ensemble.py

```diff
@@ -0,0 +1,207 @@
+from collections.abc import Sequence
+
+import torch
+from torch import Tensor
+
+from autoemulate.experimental.device import TorchDeviceMixin
+from autoemulate.experimental.emulators.base import (
+    Emulator,
+    GaussianEmulator,
+    PyTorchBackend,
+)
+from autoemulate.experimental.types import (
+    DeviceLike,
+    GaussianLike,
+    TensorLike,
+    TuneConfig,
+)
+
+
+class Ensemble(GaussianEmulator):
+    """
+    Ensemble emulator that aggregates multiple Emulator instances and returns
+    a MultivariateNormal representing the ensemble posterior.
+    Note that an Emulator instance may also be an Ensemble itself.
+
+    Parameters
+        Emulators: list[Emulator] List of fitted emulator instances.
+        Each member's `predict(x: Tensor)`
+        must return either:
+            - TensorLike (treated as mean)
+            - MultivariateNormal (with `.mean` and `.covariance_matrix`)
+    """
+
+    def __init__(
+        self,
+        emulators: list[Emulator],
+        device: DeviceLike | None = None,
+    ):
+        assert isinstance(emulators, Sequence)
+        for e in emulators:
+            assert isinstance(e, Emulator)
+        self.emulators = list(emulators)
+        self.is_fitted_ = all(e.is_fitted_ for e in emulators)
+        TorchDeviceMixin.__init__(self, device=device)
+
+    @staticmethod
+    def is_multioutput() -> bool:
+        return True
+
+    @staticmethod
+    def get_tune_config() -> TuneConfig:
+        return {}
+
+    def _fit(self, x: TensorLike, y: TensorLike) -> None:
+        for e in self.emulators:
+            e.fit(x, y)
+        self.is_fitted_ = True
+
+    @torch.inference_mode()
+    def _predict(self, x: Tensor) -> GaussianLike:
+        """
+        Perform inference with the ensemble.
+
+        Inputs:
+            - x of shape (batch_size, n_dims)
+
+        Returns:
+            - torch.distributions.MultivariateNormal with
+                - mean of shape (batch_size, n_dims)
+                - cov of shape (batch_size, n_dims, n_dims)
+        """
+
+        # Inference mode to disable autograd computation graph
+        device = x.device
+        means: list[Tensor] = []
+        covs: list[Tensor] = []
+
+        # Outputs from each emulator
+        for e in self.emulators:
+            out = e.predict(x)
+            if isinstance(out, GaussianLike):
+                mu_i = out.mean.to(device)  # (batch_size, n_dims)
+                sigma_i = out.covariance_matrix.to(
```

Comment:
For the pyright issue:
```suggestion
                assert isinstance(out.covariance_matrix, TensorLike)
                sigma_i = out.covariance_matrix.to(
```
Not sure the reason for the difference locally and the in the CI.